CRM/MembershipExtras/Hook/Pre/MembershipPaymentPlanProcessor/Contribution.php

```diff
@@ -116,48 +210,41 @@ private function alterFirstContributionParameters() {
     $this->params['contribution_recur_id'] = $this->recurringContribution['id'];
     $this->params['total_amount'] = $this->recurringContribution['amount'];
     $this->params['net_amount'] = $this->recurringContribution['amount'];
-
-    if (empty($this->params['tax_amount'])) {
-      return;
-    }
-
-    if (empty($this->fixedPeriodMembershipTypes)) {
-      $this->params['tax_amount'] = $this->calculateInstalmentTax($this->params['total_amount'], $this->params['financial_type_id']);
-
-      return;
-    }
-
-    $instalmentCount = $this->getInstalmentCountForFixedMembeship($this->fixedPeriodMembershipTypes[0]);
-    if ($this->isUsingPriceSet()) {
-      $instalmentAmount = $this->getProRatedInstalmentAmount($this->fixedPeriodMembershipTypes);
-      if ($this->isPaidMonthly()) {
-        $this->params['tax_amount'] = $this->calculateSingleInstalmentAmount($instalmentAmount->getCalculator()->getTaxAmount(), $instalmentCount);
-      }
-      else {
-        $this->params['tax_amount'] = $this->calculateSingleInstalmentAmount($instalmentAmount->getCalculator()->getTaxAmount());
-      }
-    }
-    else {
-      $this->params['tax_amount'] = $this->calculateSingleInstalmentAmount($this->params['tax_amount'], $instalmentCount);
-    }
-
+    $this->params['tax_amount'] = $this->instalmentAmount->getTaxAmount();
   }
 
   /**
-   * Gets only fixed period membeship type that belong to each line item
+   * Assigns membership types from line items.
+   * Assigns non membership price field values from line item.
+   * Sets isFixedMembershipType property
+   *
+   * @throws CRM_Core_Exception
    */
-  private function getLineItemsFixedPeriodMembershipType() {
-    $membershipTypes = [];
+  private function setMembershipTypes() {
```

Comment:
Code becomes complicated.
Should we refactor and create another 3 methods ? e.g.
- `setMembershipTypes()` should only set membershipTypes.
- `nonMembershipTypeLineItems(()`
- `adjustMinimumFee()` should run after the `setMembershipTypes()` in the `__construct`
- `setIsFixedMembershipType()`
Or maybe rename the method to a better name to reflect what is going on.

Reply:
@ahed-compucorp I have refactored the function by separating each responsibility to each function. I did not consider running method after method on the contractor as we would need to iterate line items each time.